scipy/optimize/_numdiff.py

```diff
@@ -0,0 +1,297 @@
+    h_adjusted[central_adjusted] = np.minimum(lower_dist[central_adjusted],
+                                              upper_dist[central_adjusted])
+    return h_adjusted, scheme
+
+
+def _approx_forward(f, x0, h, f0, args, bounds):
+    h = _adjust_step_1st_order(x0, h, bounds)
+
+    if f0 is None:
+        f0 = f(x0, *args)
+
+    if x0.ndim == 0:
+        x = x0 + h
+        dx = x - x0
+        df = f(x, *args) - f0
+        J = df / dx
+    else:
+        J = []
+        for i in range(x0.size):
+            x = np.copy(x0)
+            x[i] += h[i]
+            dx = x[i] - x0[i]
+            df = f(x, *args) - f0
+            J.append(df / dx)
+        J = np.array(J).T
+    return J
+
+
+def _approx_central(f, x0, h, f0, args, bounds):
+    h, scheme = _adjust_step_2nd_order(x0, h, bounds)
+
+    if np.any(scheme == 1) and f0 is None:
+        f0 = f(x0, *args)
+
+    if x0.ndim == 0:
+        if scheme == 0:
+            x1 = x0 - h
+            x2 = x0 + h
+            dx = x2 - x1
+            f1 = f(x1, *args)
+            f2 = f(x2, *args)
+            J = (f2 - f1) / dx
+        else:
+            x1 = x0 + h
+            x2 = x0 + 2 * h
+            dx = x2 - x0
+            f1 = f(x1, *args)
+            f2 = f(x2 * h, *args)
+            J = (-3 * f0 + 4 * f1 - f2) / dx
+    else:
+        J = []
+        for i in range(x0.size):
+            x1 = np.copy(x0)
+            x2 = np.copy(x0)
+            if scheme[i] == 0:
+                x1[i] -= h[i]
+                x2[i] += h[i]
+                dx = x2[i] - x1[i]
+                f1 = f(x1, *args)
+                f2 = f(x2, *args)
+                J.append((f2 - f1) / dx)
+            else:
+                x1[i] += h[i]
+                x2[i] += 2 * h[i]
+                f1 = f(x1, *args)
+                f2 = f(x2, *args)
+                dx = x2[i] - x0[i]
+                J.append((-3 * f0 + 4 * f1 - f2) / dx)
+        J = np.array(J).T
+
+    return J
+
+
+def approx_derivatives(f, x0, method='central', rel_step=None,
+                       f0=None, args=(), bounds=(None, None)):
+    """Compute a finite difference approximation of the derivatives of a
+    vector-valued function.
+
+    If a function maps from R^n to R^m, its derivatives form m x n matrix
+    called Jacobian, where an element (i, j) is equal to the partial derivative
+    of f[i] with respect to x[j].
+
+    Parameters
+    ----------
+    f : callable
+        The function of which to estimate the derivatives. Takes and returns
+        array_like.
+    x0 : array_like
+        The point at which to estimate the derivatives.
+    method : {'central', 'forward'}, optional
+        Finite difference scheme to use.
+    rel_step : None or array_like, optional
+        Relative step size. The actual step size is computed as
+        ``h[i] = rel_step[i] * max(1, abs(x[i])``. If None, it is set to
+        eps**0.5 for `method`="center" and eps**(1/3) for `method`="forward",
```

Comment:
asterisks have special meaning for Sphinx, hence need to be shielded by eg double backticks. Ideally, build the docs locally and make sure the markup looks OK. (one way would be `python runtests --shell; cd <scipy-source-dir>/doc; make html-scipyorg`)